File: libgomp/testsuite/libgomp.fortran/task-detach-12.f90
program test
    use omp_lib
    implicit none
    integer(omp_event_handle_kind) :: oevent, ievent
    integer :: i
    integer, allocatable :: temp(:)
    ALLOCATE(temp(5))

    !$omp parallel num_threads(3)
    !$omp single
    DO i=1,5
    !$omp task firstprivate(i) firstprivate(temp)  detach(oevent)
          temp(:) = 0;
          temp(1) = -1;
          !print *,temp
          call omp_fulfill_event(oevent)
    !$omp end task
    ENDDO
    !$omp taskwait
    !$omp end single
    !$omp end parallel
end program

File: libgomp/testsuite/libgomp.c-c++-common/task-detach-12.c
/* { dg-do run } */
/* { dg-options "-fopenmp" } */

#include <omp.h>

int
main ()
{
  struct S { int a[7]; } s = { { 1, 2, 3, 4, 5, 6, 7 } };
  omp_event_handle_t x;
  #pragma omp parallel master
  #pragma omp task firstprivate (s) detach (x)
    {
      if (s.a[3] != 4)
	__builtin_abort ();
      omp_fulfill_event (x);
    }
  return 0;
}
